globus_automate_client/cli/callbacks.py

```diff
@@ -85,7 +85,7 @@ def text_validator_callback(message: str) -> str:
     return message
 
 
-def principal_validator_callback(principals: List[str]) -> List[str]:
+def principal_validator(principals: List[str]) -> List[str]:
```

Comment:
Can we just make something like `special_vals` a kwarg here that defaults to an empty set. Then, as we iterate check for presence in the `special_vals` set or having the proper principal id format here.

Then, the specialized functions can implement in terms of this by providing the kwarg.